src/libcore/lib.rs

```diff
@@ -85,6 +85,7 @@
 #![feature(unwind_attributes)]
 #![cfg_attr(stage0, feature(simd))]
 #![cfg_attr(not(stage0), feature(repr_simd, platform_intrinsics))]
+#![feature(slice_patterns)]
```

Comment:
We've historically had a lot of codegen issues with this in the past, so I'm somewhat wary to enable this to implement this for now. Can this be implemented without this feature?